lib/kafka/connection.rb

```diff
@@ -75,25 +75,27 @@ def close
     #
     # @return [Object] the response.
     def send_request(request)
-      # Default notification payload.
-      notification = {
-        broker_host: @host,
-        api: Protocol.api_name(request.api_key),
-        request_size: 0,
-        response_size: 0,
-      }
+      with_retry do
+        # Default notification payload.
+        notification = {
+          broker_host: @host,
+          api: Protocol.api_name(request.api_key),
+          request_size: 0,
+          response_size: 0,
+        }
 
-      @instrumenter.instrument("request.connection", notification) do
-        open unless open?
+        @instrumenter.instrument("request.connection", notification) do
+          open unless open?
 
-        @correlation_id += 1
+          @correlation_id += 1
 
-        write_request(request, notification)
+          write_request(request, notification)
 
-        response_class = request.response_class
-        wait_for_response(response_class, notification) unless response_class.nil?
+          response_class = request.response_class
+          wait_for_response(response_class, notification) unless response_class.nil?
+        end
       end
-    rescue Errno::EPIPE, Errno::ECONNRESET, Errno::ETIMEDOUT, EOFError => e
+    rescue Errno::ETIMEDOUT => e
```

Comment:
@dasch "safe" in the sense that doing this could cause an unexpected error? or something else?